src/Features/CSharp/Portable/CodeFixes/UseImplicitTyping/UseExplicitTypingCodeFixProvider.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Simplification;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.UseImplicitTyping
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.UseExplicitTyping), Shared]
+    internal class UseExplicitTypingCodeFixProvider : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds =>
+            ImmutableArray.Create(IDEDiagnosticIds.UseExplicitTypingDiagnosticId);
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var span = context.Span;
+            var root = await document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            var token = root.FindToken(span.Start);
+            if (!token.Span.IntersectsWith(span))
+            {
+                return;
+            }
+
+            var node = token.GetAncestors<SyntaxNode>().First(n => n.Span.Contains(span));
+            if (node == null ||
+                !(node.IsKind(SyntaxKind.PredefinedType) ||
+                  node.IsKind(SyntaxKind.ArrayType) ||
+                  node.IsKind(SyntaxKind.IdentifierName) ||
+                  node.IsKind(SyntaxKind.GenericName) ||
+                  node.IsKind(SyntaxKind.AliasQualifiedName)))
+            {
```

Comment:
```C#
if (node?.MatchesKind(SyntaxKind.PredefinedType, SyntaxKind.ArrayType, SyntaxKind.IdentifierName, SyntaxKind.GenericName, SyntaxKind.AliasQualifiedName) == false)
{
    return false;
}
```

Note that this would create a params array, so you might want to lift that out.